components/lib/fetch-httpurlconnection/src/main/java/mozilla/components/lib/fetch/httpurlconnection/HttpURLConnectionClient.kt

```diff
@@ -62,6 +74,14 @@ private fun HttpURLConnection.setupWith(request: Request) {
     request.readTimeout?.let { (timeout, unit) ->
         readTimeout = unit.toMillis(timeout).toInt()
     }
+
+    val cookieManager = getOrCreateCookieManager()
+    if (request.cookiePolicy == Request.CookiePolicy.OMIT) {
+        val uri = URL(request.url).toURI()
+        for (cookie in cookieManager.cookieStore.get(uri)) {
+            cookieManager.cookieStore.remove(uri, cookie)
+        }
```

Comment:
Let's maybe add a comment here why we are doing this - because it looks weird if you do not know the background. :)

Reply:
OK, comment added.